spec/update-process-env-spec.js

```diff
@@ -57,31 +60,60 @@ describe('updateProcessEnv(launchEnv)', function () {
         ATOM_HOME: '/the/atom/home'
       }
 
-      updateProcessEnv({PWD: '/the/dir'})
+      updateProcessEnv({ATOM_SUPPRESS_ENV_PATCHING: 'true', PWD: '/the/dir'})
       expect(process.env).toEqual({
         PWD: '/the/dir',
+        ATOM_SUPPRESS_ENV_PATCHING: 'true',
         NODE_ENV: 'the-node-env',
         NODE_PATH: '/the/node/path',
         ATOM_HOME: '/the/atom/home'
       })
 
-      updateProcessEnv({PWD: '/the/dir', ATOM_HOME: path.join(newAtomHomePath, 'non-existent')})
+      updateProcessEnv({ATOM_SUPPRESS_ENV_PATCHING: 'true', PWD: '/the/dir', ATOM_HOME: path.join(newAtomHomePath, 'non-existent')})
       expect(process.env).toEqual({
+        ATOM_SUPPRESS_ENV_PATCHING: 'true',
         PWD: '/the/dir',
         NODE_ENV: 'the-node-env',
         NODE_PATH: '/the/node/path',
         ATOM_HOME: '/the/atom/home'
       })
 
-
-      updateProcessEnv({PWD: '/the/dir', ATOM_HOME: newAtomHomePath})
+      updateProcessEnv({ATOM_SUPPRESS_ENV_PATCHING: 'true', PWD: '/the/dir', ATOM_HOME: newAtomHomePath})
       expect(process.env).toEqual({
+        ATOM_SUPPRESS_ENV_PATCHING: 'true',
         PWD: '/the/dir',
         NODE_ENV: 'the-node-env',
         NODE_PATH: '/the/node/path',
         ATOM_HOME: newAtomHomePath
       })
     })
+
+    it('allows ATOM_SUPPRESS_ENV_PATCHING to be preserved if set', function () {
```

Comment:
I see now, if we *ever* launched from a shell, we'll never shell out for the environment going forward. Makes sense. Disregard above comment.